DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/AddDelegateToGroupController.cs

```diff
@@ -0,0 +1,151 @@
+        public IActionResult Index(
+            int groupId = 1,
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
+        {
+            sortBy ??= DefaultSortByOptions.Name.PropertyName;
+            filterBy = FilteringHelper.GetFilterBy(
+                filterBy,
+                filterValue,
+                Request,
+                AddDelegateToGroupCookieName,
+                DelegateActiveStatusFilterOptions.IsActive.FilterValue
+            );
+
+            var centreId = User.GetCentreId();
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(centreId);
+            var delegateUsers = userDataService.GetDelegatesNotRegisteredForGroupByGroupId(groupId, centreId);
+            var groupName = userDataService.GetGroupNameById(groupId);
+
+            var model = new AddDelegateToGroupViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                page,
+                groupId,
+                groupName!,
+                searchString,
+                sortBy,
+                sortDirection,
+                filterBy
+            );
+
+            Response.UpdateOrDeleteFilterCookie(AddDelegateToGroupCookieName, filterBy);
+            return View(model);
+        }
+
+        [Route("DelegateAddedToGroupConfirmation/{groupId:int}/{delegateId:int}")]
+        public IActionResult DelegateAddedToGroupConfirmation(int groupId, int delegateId)
+        {
+            var delegateUser = userDataService.GetDelegateUserById(delegateId);
+            if (delegateUser is null)
+            {
+                return NotFound();
+            }
+
+            var adminId = User.GetAdminId();
+            var groupName = userDataService.GetGroupNameById(groupId);
+
+            var newDetails = new MyAccountDetailsData(
+                adminId,
+                delegateUser.Id,
+                string.Empty,
+                delegateUser.FirstName!,
+                delegateUser.LastName!,
+                delegateUser.EmailAddress!,
+                null
+            );
+
+            groupsDataService.AddDelegateToGroup(
+                delegateUser!.Id,
+                groupId,
+                clockService.UtcNow,
+                0
+            );
+
+            groupsService.EnrolDelegateOnGroupCourses(
+                delegateUser!,
+                newDetails,
+                groupId,
+                adminId
+            );
+
+            var model = new DelegateAddedToGroupConfirmationViewModel(delegateUser.FullName, groupName!, groupId);
+            return View(model);
+        }
+
+        [Route("AddDelegateToGroupItems/{groupId:int}")]

```

Comment:
Following the comment about the main controller route, this route can probably change too:
`AllItems`

Reply:
This has now been refactored as per the original comment.